### PR TITLE
Ember 2.0.3 Support

### LINKED ISSUE #18: Ember 2 compatibility
- `23) RestModel.V2 ::request when it is a GET request and the request is cached and the response is an array "before each" hook:
   TypeError: MutatingArray.apply(...).set(...).runFilters is not a function`
- `DEPRECATION: Ember.String.fmt is deprecated, use ES6 template strings instead. [deprecation id: ember-string-utils.fmt] See https://babeljs.io/docs/learn-es6/#template-strings for more details.`

### PR DESCRIPTION
**Pull request by Dhaulagiri**
Updates rest-model to support ember 2.0.

Fixes #18

### CODE REVIEW

**Comment by Dhaulagiri** [comment]
@heroku/hit this is ready for review

File: dist/rest-model.js [hunk 1]
```diff
@@ -598,15 +595,15 @@ module.exports = Ember.Object.extend({
...
-    this.getParentKeyNames().forEach(function(key) {
-      var parent     = parents[key];
+  assertHasParentKeys: function assertHasParentKeys(parents) {
+    this.getParentKeyNames().forEach((function (key) {
+      var parent = parents[key];
       var primaryKey = parent ? this.getPrimaryKey(parent) : null;
 
       if (Ember.isNone(primaryKey)) {
-        throw new Error('No primary key found for parent "%@".'.fmt(key));
+        throw new Error('No primary key found for parent "' + key + '".');
```
Comment (jmonster) [inline dist/rest-model.js:604]: `use ${positiveAdjective} generator strings?`

File: dist/rest-model.js [hunk 2]
```diff
@@ -1211,8 +1209,8 @@ module.exports = Ember.Object.extend({
    * @param {String,Number} key the primary key to build the cache key for
    * @return {String} a cache key
    */
-  getCacheKey: function(klass, key) {
-    return '%@: %@'.fmt(klass, key);
+  getCacheKey: function getCacheKey(klass, key) {
+    return klass + ': ' + key;
```
Comment (jmonster) [inline dist/rest-model.js]: generator string

File: dist/rest-model.js (hunk 1 above)
Comment (Dhaulagiri) [inline dist/rest-model.js:604]: I'm not sure what you mean?

**Comment by jmonster** [comment]
:+1:

**Comment by jmonster** [comment]
> You may fire when ready

**Pull request merged**